plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundReadWrite.java

```diff
@@ -71,53 +138,80 @@ static GroundReadWrite getInstance() {
+            LOG.info("client cass is " + clientClass);
             if (TEST_CONN.equals(clientClass)) {
-                conn = testConn;
-                LOG.debug("Using test connection.");
+                setConn(testConn);
+                LOG.info("Using test connection."); // for unit and integration test
+                dbClient = new PostgresClient("127.0.0.1", 5432, "test", "test", "test");
+                createInstance();
             } else {
+                conf.set("edu.berkeley.ground.model.config.clientClass", clientClass);
+                conf.set("edu.berkeley.ground.model.config.host", host);
+                conf.set("edu.berkeley.ground.model.config.dbName", dbName);
+                conf.set("edu.berkeley.ground.model.config.userName", userName);
+                conf.set("edu.berkeley.ground.model.config.password", password);
+                conf.setInt("edu.berkeley.ground.model.config.port", port);
+                Class<?> clazz = Class.forName(clientClass);
+                Constructor<?> constructor = clazz.getConstructor(String.class, Integer.class,
+                        String.class, String.class, String.class);
+                dbClient = (DBClient) constructor.newInstance(host, port, dbName, userName, password);
+                // dbClient = new PostgresClient(host, port, dbName, userName, password);
                 LOG.debug("Instantiating connection class " + clientClass);
-                Object o = createInstance(clientClass);
-                if (DBClient.class.isAssignableFrom(o.getClass())) {
-                    dbClient = (DBClient) o;
-                    conn = dbClient.getConnection();
-                } else {
-                    throw new IOException(clientClass + " is not an instance of DBClient.");
-                }
-                o = createInstance(graphFactoryType);
-                if (GraphFactory.class.isAssignableFrom(o.getClass())) {
-                    graphFactory = (GraphFactory) o;
-                }
-                o = createInstance(nodeFactoryType);
-                if (NodeFactory.class.isAssignableFrom(o.getClass())) {
-                    nodeFactory = (NodeFactory<?>) o;
-                }
-                o = createInstance(edgeFactoryType);
-                if (EdgeFactory.class.isAssignableFrom(o.getClass())) {
-                    edgeFactory = (EdgeFactory) o;
-                }
+                createInstance();
             }
         } catch (Exception e) {
             throw new RuntimeException(e);
         }
     }
 
-    private Object createInstance(String clientClass)
-            throws ClassNotFoundException, InstantiationException, IllegalAccessException {
-        Class<?> c = Class.forName(clientClass);
-        Object o = c.newInstance();
-        return o;
-    }
+    private void createInstance() throws GroundDBException {
+        PostgresVersionSuccessorFactory succ = new PostgresVersionSuccessorFactory();
```

Comment:
At this point we have tested with Postgres. We have not tested with Neo4j. I have not been able to setup my environment to run with Neo4j yet.